src/serialize.h

```diff
@@ -124,33 +134,19 @@ template<typename Stream> inline uint64_t ser_readdata64(Stream &s)
     s.read((char*)&obj, 8);
     return le64toh(obj);
 }
-inline uint64_t ser_double_to_uint64(double x)
-{
-    static_assert(sizeof(uint64_t) == sizeof(double), "Floating-point width assumption");
-    uint64_t y;
-    memcpy(&y, &x, sizeof(y));
-    return y;
-}
-inline uint32_t ser_float_to_uint32(float x)
-{
-    static_assert(sizeof(uint32_t) == sizeof(float), "Floating-point width assumption");
-    uint32_t y;
-    memcpy(&y, &x, sizeof(y));
-    return y;
-}
-inline double ser_uint64_to_double(uint64_t y)
+template<typename Stream> inline float ser_readdata_float(Stream &s)
 {
-    static_assert(sizeof(double) == sizeof(uint64_t), "Floating-point width assumption");
-    double x;
-    memcpy(&x, &y, sizeof(x));
-    return x;
+    static_assert(sizeof(float) == 4, "Floating-point width assumption");
+    float obj;
+    s.read((char*)&obj, 4);
```

Comment:
Will this not read a corrupted value on files saved (before this change) on big endian systems (ie: previously saved using little endian but now interpreted using big endian?) - I'm not overly familiar with the code so maybe it's okay - but just something that jumps out.

Reply:
@donaloconnor Floating point number serialization in memory is specified by IEEE 754, IIRC, and not subject to the endianness under which integers are encoded.

Reply:
@sipa - I was under the impression that endianess is not specified in IEEE 754 - but subject to it as it's implementation dependant. This is probably not an issue in practice as BE archs are pretty rare now. My point was that the byte ordering might be switched when loading the file on a BE float arch. I've read that even some systems have half endianess where integer are LE while 754 are BE. Anyway that is a rabbit hole and maybe out of scope here. Thanks for the response.

Reply:
Ah, that's good to know in any case. If true, that means this PR doesn't really add anything...

Reply:
@sipa Should I revert to the version I originally submitted that solves the UB part and the UB part only? :-)

Reply:
@practicalswift Yes, that makes sense. I wasn't aware that IEEE encoding was still subject to BE/LE serialization.

Ideally we actually test this on some BE hardware, though...